CommonItem/src/test/java/com/github/sef24sp4/common/item/loottable/LootTableTest.java

```diff
@@ -0,0 +1,102 @@
+package com.github.sef24sp4.common.item.loottable;
+
+import com.github.sef24sp4.common.item.CommonItem;
+import com.github.sef24sp4.common.item.ItemRarity;
+import com.github.sef24sp4.common.item.ItemSPI;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class LootTableTest {
+	private LootTable lootTable;
+	private Map<ItemRarity, Double> rarityChances;
+
+	private ItemSPI mockItemSPI;
+	private CommonItem commonItem;
+
+	@BeforeEach
+	void setUp() {
+		this.rarityChances = new HashMap<>();
+		this.lootTable = new LootTable(this.rarityChances);
+		this.mockItemSPI = mock(ItemSPI.class);
+		this.commonItem = mock(CommonItem.class);
+	}
+
+	@Test
+	void testConstructorWithInvalidArguments() {
+		this.rarityChances.put(ItemRarity.COMMON, -0.1);
+		assertThrows(IllegalArgumentException.class, () -> new LootTable(this.rarityChances));
+		this.rarityChances.put(ItemRarity.UNCOMMON, 1.1);
+		assertThrows(IllegalArgumentException.class, () -> new LootTable(this.rarityChances));
+		this.rarityChances.put(ItemRarity.RARE, 0.6);
+		assertThrows(IllegalArgumentException.class, () -> new LootTable(this.rarityChances));
+	}
```

Comment:
Subsequent tests are currently being "shadowed" by the first test case. 
In other words `IllegalArgumentException` is not necessarily thrown because of the new test case, but because the map is polluted with an illegal state from the previous test. 

This is best illustrated with what is going on line 38-39. 
In an isolated environment this should not throw an exception. However, an exception is still thrown because the map still contains illegal values. 

Consider using [Parameterized tests](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests).

Something like: 
```java
/**
 * Argument source for {@link #testConstructorWithInvalidArguments(Map<ItemRarity, Double>)}.
 *
 * @return The test arguments.
 * @see ParameterizedTest
 * @see MethodSource
 */
public static Stream<Arguments> testConstructorWithInvalidArguments() {
	return Stream.of(
			Arguments.of(Map.ofEntries( // TEST CASE 1
					Map.entry(ItemRarity.COMMON, 1.2)
			)),
			Arguments.of(Map.ofEntries( // TEST CASE 2
					Map.entry(ItemRarity.COMMON, 0.5),
					Map.entry(ItemRarity.RARE, -0.1)
			)),
			Arguments.of(Map.ofEntries( // TEST CASE 4
					Map.entry(ItemRarity.UNCOMMON, 0.2),
					Map.entry(ItemRarity.COMMON, 0.5),
					Map.entry(ItemRarity.RARE, 0.4)
			))
	);
}

@ParameterizedTest
@MethodSource
public void testConstructorWithInvalidArguments(final Map<ItemRarity, Double> map) {
	// ... code
}
```

Consider also making a parameterized test for cases where the constructor should not fail. Can use the `assertDoesNotThrow`. 

> [!TIP]
> Real world **Example**
> https://github.com/SE-F24-SP4/semesterproject-4/blob/2739d551cb1c0fe6bb9c31c71f521408c7ec0436/CollisionSystem/src/test/java/com/github/sef24sp4/collisionsystem/map/BucketTest.java#L222-L227
> Example of the corresponding method source. Where the arguments are provided. 
> https://github.com/SE-F24-SP4/semesterproject-4/blob/2739d551cb1c0fe6bb9c31c71f521408c7ec0436/CollisionSystem/src/test/java/com/github/sef24sp4/collisionsystem/map/BucketTest.java#L81-L97